OlinkAnalyze/vignettes/bridging_E3072toEHT.Rmd

```diff
@@ -0,0 +1,682 @@
+normalization using the median of paired differences. The result is an
+adjusted Explore 3072 data set with three additional columns:
+
+-   `BridgingRecommendation`: a flag which indicates if the assay is
+    bridgeable and, if so, which normalization method is recommended
+
+-   `MedianCenteredNPX`: NPX values after normalization using the median
+    of paired differences
+
+-   `QSNormalizedNPX`: NPX values after normalization using quantile
+    smoothing
+    
+Note that regardless of the bridging recommendation, NPX values will be
+available for both normalization methods. A visual representation of the between-product bridging workflow is shown below.
+
+```{r, fig.cap= fcap, eval = TRUE, echo = FALSE, out.width="100%"}
+knitr::include_graphics(normalizePath("../man/figures/Bridging_schematic.png"), error = FALSE)
+fcap <- "Schematic of Explore 3072 to Explore HT Bridging Workflow"
+```
+
+## Import NPX files
+
+To normalize Explore 3072 data to Explore HT data, first the two data
+sets are read into R using `read_NPX()`. If more than two data sets are
+being normalized, all Explore 3072 studies should be normalized together
+prior to normalizing between products and the concatenated bridged data
+set should be used as the input. In the case of multiple Explore HT
+studies, only one Explore HT study should be chosen as the reference
+data set. The data can be loaded using `read_NPX()` function with
+default Olink Software NPX file as input, as shown below.
+
+```{r message=FALSE, eval=FALSE, echo = TRUE}
+data_E3072 <- read_NPX("~/NPX_Explore3072_location.parquet") # Could also be a CSV file
+data_EHT <- read_NPX("~/NPX_ExploreHT_location.parquet")
+```
+
+## Checking input datasets and bridging samples
+
+First, confirm that there are overlapping sample IDs within the study.
+In a case where the same sample have different sample IDs in the two projects, the `overlap_sample_list`
+will contain 2 arrays of equal length where the index of each entry
+corresponds to the same sample. For example, if a sample had the
+SampleID of `Sample_1_Aliquot_1` in the first project and
+`Sample_1_Aliquot_2` in the second project, then the overlap sample list
+should look as the following.
+
+```{r eval=FALSE, include=TRUE}
+overlap_sample_list <-list("DF1" = c("A1", "A2", "A3", "Sample_1_Aliquot_1"),
+                           "DF2" = c("A1", "A2", "A3",  "Sample_1_Aliquot_2")) 
+```
+
+Note that external controls should not be included in the list of
+bridging samples, as detailed in the [Bridge Sample Selection] section
+of this tutorial. External control samples often share the same naming
+convention across data sets but may represent different samples due to
+reagent batch differences. Appending the project name to the end of the
+control samples can ensure unique Sample IDs.
+
+```{r, echo=TRUE, eval=FALSE}
+data.frame(SampleID = intersect(data_E3072$SampleID, data_EHT$SampleID)) |>
```

Comment:
Per D8493, it looks like Plate Controls, Negative Controls, Sample Controls, and Samples are output in the standard parquet format. Would it be more applicable to change the filtering criteria to dplyr::filter(SampleType == "SAMPLE")?

Reply:
Also not sure if it's better to use a regex for handling CSV files - example in import NPX file indicates that 3K CSV files are also valid input

Reply:
If we keep in str_detect, should we use the stringr::str_detect convention to match the code below? Should we add dplyr::filter and dplyr::mutate in the rbind section?

Reply:
I changed all to SampleType (and AssayType) and added a note if these columns arent present.